ui/app/components/sidebar/frame.js

```diff
@@ -7,8 +7,15 @@ import Component from '@glimmer/component';
 import { service } from '@ember/service';
 import { inject as controller } from '@ember/controller';
 
+import { TOGGLE_WEB_REPL } from 'vault/utils/analytic-events';
+
 export default class SidebarNavComponent extends Component {
+  @service analytics;
   @service currentCluster;
   @service console;
   @controller('vault.cluster') clusterController;
+
+  trackReplToggle = () => {
+    this.analytics.trackEvent(TOGGLE_WEB_REPL);
+  };
```

Comment:
This is an example of how to track custom events in the code. 

We could also do something like `{{on "click" (fn this.analytics.trackEvent MY_EVENT)}}` in the template, but I think the component file is a little cleaner.

Reply:
Yeah! The plan is add event tracking to the toolkit for new features as we release them. It's something we can discuss with product and design as well. "How do we know this feature is successful?" can be answered with the data coming back from this work. 

The first thing we're aiming to track is the reporting work, but all new features (and probably many old one) can be added as devs see fit!